metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -857,33 +860,40 @@ class MetalsLanguageServer(
   def didChangeWatchedFiles(
       event: DirectoryChangeEvent
   ): CompletableFuture[Unit] = {
-    val path = AbsolutePath(event.path())
-    val isScalaOrJava = path.isScalaOrJava
-    if (isScalaOrJava && event.eventType() == EventType.DELETE) {
-      diagnostics.didDelete(path)
-      CompletableFuture.completedFuture(())
-    } else if (isScalaOrJava && !savedFiles.isRecentlyActive(path)) {
-      event.eventType() match {
-        case EventType.CREATE =>
-          buildTargets.onCreate(path)
-        case _ =>
-      }
-      onChange(List(path)).asJava
-    } else if (path.isSemanticdb) {
+    if (event.eventType() == EventType.OVERFLOW && event.path() == null) {
       CompletableFuture.completedFuture {
```

Comment:
OVERFLOW event usually had null as path, so it was never addressed. I tested it by creating 1k files automatically.